server/server.go

```diff
@@ -608,6 +603,13 @@ func (s *Server) createClient(conn net.Conn) *client {
 		c.mu.Lock()
 	}
 
+	// Check for Auth. We schedule this timer after the TLS handshake to avoid
+	// the race where the timer fires during the handshake and causes the
+	// server to write bad data to the socket. See issue #432.
+	if authRequired {
```

Comment:
You could move this after the check for `c.nc == nil` since connection may have been closed while lock was released in TLS case above.